File: src/main/java/com/ByTech/ByTech/repositories/IUserRepository.java
package com.ByTech.ByTech.repositories;

import com.ByTech.ByTech.models.UserModel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;


@Repository
public interface IUserRepository extends JpaRepository<UserModel,Long> {
    Optional<UserModel> findByUserNameAndPassword(String userName, String password);

}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/User/repositories/IUserRepository.kt
package com.example.ByTech_movil.User.repositories

import com.example.ByTech_movil.User.models.UserModel
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface IUserRepository : JpaRepository<UserModel?, Long?> {
    fun findByUserNameAndPassword(userName: String?, password: String?): Optional<UserModel?>?
    fun findByUserName(userName: String?): Optional<UserModel?>?
}
